homeassistant/components/wallbox/number.py

```diff
@@ -35,11 +32,11 @@ class WallboxNumber(CoordinatorEntity, NumberEntity):
     def __init__(self, coordinator, config):
         """Initialize a Wallbox sensor."""
         super().__init__(coordinator)
-        _properties = CONF_SENSOR_TYPES[CONF_MAX_CHARGING_CURRENT_KEY]
+        sensor_description = SENSOR_TYPES[CONF_MAX_CHARGING_CURRENT_KEY]
```

Comment:
The number platform is not allowed to use a sensor entity description. It should have its own description.

Reply:
Technically only `name` and `device_class` of the description are accessed, but I agree that this isn't ideal and should be changed. As I understand it `CONF_MAX_CHARGING_CURRENT_KEY` is currently exposed as both a `sensor` and `number`. Maybe it would make sense to only keep the `number` entity for it going forward.

@hesselonline What do you think? Doing so would likely be a breaking change.

Reply:
Yes, I understand the point of not wanting to reuse SensorEntityDescription in the number component. I would create a seperate dict in const.py name NUMBER_TYPES, containing NumberEntityDescription. I would then move the `filtered_data = {k: data[k] for k in SENSOR_TYPES if k in data}` part to the number and sensor components. I would like to keep the `CONF_MAX_CHARGING_CURRENT` sensor as not all Wallbox users are authorized to use the number component.

Reply:
The filtering can just be removed. Sounds good! 👍